src/sql/workbench/contrib/query/browser/gridPanel.ts

```diff
@@ -388,7 +390,7 @@ export abstract class GridTableBase<T> extends Disposable implements IView {
 		super();
 
 		this.options = { ...defaultGridTableOptions, ...options };
-
+		this.copyHeaders = this.configurationService.getValue<boolean>('queryEditor.results.saveAsCsv.includeHeaders');
```

Comment:
did you mean to use the configuration mentioned in the issue 'queryEditor.results.copyIncludeHeaders'?

Reply:
I think this setting used in the PR is only for exporting results to CSV.  The copy behavior would likely be a different setting (though honestly I got the design around some of these query editor settings).

Reply:
You should query this value in the copy callbacks to ensure that you have the correct value - it currently looks like if someone runs a query and then changes the value of this setting the grid will still have the original value if they then copy the results.